lib/tasks/convert_application_yml_to_proper_types.rake

```diff
@@ -0,0 +1,66 @@
+namespace :convert_application_yml_to_proper_types do
+  desc 'Take application.yml input file and convert to proper type'
+  task :convert, [:filename] => :environment do |t, args|
+    VALUE_TYPES = ['float', 'string', 'boolean', 'json', 'timestamp', 'integer']
+    filename = args[:filename]
+
+    if filename
+      puts "Making backup copy of '#{filename}' to '#{filename}.bak'"
+      FileUtils.copy(filename, filename+'.bak')
+      content = ''
+      add_spaces = false
+      File.open(filename+'.tmp', 'w') do |newfile|
+        IO.foreach(filename) do |l|
+          add_spaces = true if l.include?('development:')
+          content=l
+          key_value = l.split(':')
+
+          if key_value.length == 2 && !key_value[1].strip.blank?
+            key = key_value[0].strip
+            value = key_value[1].strip
+            value = value.strip
+            value_type = find_value_type(key)
+            converted_val = convert(value, value_type)
+            content = "#{key}: #{converted_val}"
+            content = "  #{key}: #{converted_val}" if add_spaces
+          end
+
+          newfile.puts content
+        end
+      end
+      FileUtils.mv filename+'.tmp', filename
+    else
+      puts 'Please specify argument of filename.'
+      puts 'Example: rake convert_application_yml_to_proper_types:convert[application.yml]'
+    end
+  end
+
+  def find_value_type(key)
+    IO.foreach("#{Rails.root}/lib/identity_config.rb") do |l|
+      if l.include?(key)
+        VALUE_TYPES.each_with_index { |word, index|
+           return VALUE_TYPES[index] if l.index(word)
+        }
+      end
+
+      nil
+    end
+  end
```

Comment:
maybe instead of parsing `identity_config` could we update it so that it stores the type information and we can just access that in-memory by loading the file?

Reply:
Yes, that's an option.  I'll update so the type information is loaded from identity_config and the it stores it in memory for processing.